### PR TITLE
[4.0][Workflow] Use new Event System for featured event and some gui fixes

### PR DESCRIPTION
**Pull request by HLeithner**
### Summary of Changes
Refactor the Feature up/down event to the new Event system and fix some gui issues for publising and feature button on articles overview and featured view.

### Testing Instructions
* Publish/Unpublish Articles with and without workflow active and with and without publishing plugin active on articles, featured and article edit view.
* Feature/Unfeature Articles with and without workflow active and with and without publishing plugin active on articles, featured and article edit view.

* Bonus track (with workflow enabled), disable the feature/publishing plugin
* go to the articles overview page
* Activate the feature/publishing plugin
* Press the feature/publishing button
* Should fail with an proper error message

### Expected result
Works

### Actual result
* Publish up/down button is not click able if workflow publishing button is not active
* Feature up/down button is not click on featured view able if workflow featuring button is not active

### CODE REVIEW

File: administrator/components/com_content/src/Controller/ArticlesController.php [hunk 1]
```diff
@@ -70,6 +70,9 @@ public function featured()
 		$values = array('featured' => 1, 'unfeatured' => 0);
 		$task   = $this->getTask();
 		$value  = ArrayHelper::getValue($values, $task, 0, 'int');
+		$view   = $this->input->get('view', '');
+
+		$redirectUrl = 'index.php?option=com_content&view=' . ($view === 'featured' ? 'featured' : 'articles');
```
Comment (wilsonge) [inline administrator/components/com_content/src/Controller/ArticlesController.php]: Think this needs the `$this->getRedirectToListAppend()`

Comment (HLeithner) [reply administrator/components/com_content/src/Controller/ArticlesController.php]: Added it.

File: administrator/components/com_content/src/Controller/ArticlesController.php [hunk 2]
```diff
@@ -70,6 +70,10 @@ public function featured()
 		$values = array('featured' => 1, 'unfeatured' => 0);
 		$task   = $this->getTask();
 		$value  = ArrayHelper::getValue($values, $task, 0, 'int');
+		$view   = $this->input->get('view', '');
+
+		$redirectUrl = 'index.php?option=com_content&view=' . ($view === 'featured' ? 'featured' : 'articles')
```
Comment (wilsonge) [inline administrator/components/com_content/src/Controller/ArticlesController.php]: Use `$this->view_list` here and no need for the `if` statement - see https://github.com/joomla/joomla-cms/pull/29265/files#diff-030e861675d06f3fa86621ee1ba5d7dfR48-R51

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**